crates/toml_writer/src/integer.rs

```diff
@@ -0,0 +1,194 @@
+use core::fmt::{self, Display};
+
+/// N must be an integer type.
```

Comment:
The first line of a docstring is generally used as a summary.  We should probably use [Display](https://doc.rust-lang.org/std/path/struct.Display.html) for inspiration.

Also considering people can't create a `TomlInteger` directly, I don't think we need to document the constraints for `N`

Reply:
I'll just remove the docstring for now.

Reply:
I think it would be good to help people know more explicitly the relationship of the types than having to read their functions to figure it out

Reply:
... says the person who didn't do that for the string stuff